generator/internal/dart/templates/lib/service.mustache

```diff
@@ -17,11 +17,19 @@ limitations under the License.
 {{#Codec.DocLines}}
 {{{.}}}
 {{/Codec.DocLines}}
-class {{Codec.Name}} extends CloudService {
-  static const String host = '{{DefaultHost}}';
+class {{Codec.Name}} {
+  static const String _host = '{{DefaultHost}}';
 
-  {{Codec.Name}}({required super.client});
+  final ServiceClient _client;
+
+  {{Codec.Name}}({required http.Client httpClient})
+      : _client = ServiceClient(httpClient: httpClient);
   {{#Codec.Methods}}
   {{> method}}
   {{/Codec.Methods}}
+
+  /// Closes the client and cleans up any resources associated with it.
+  ///
+  /// Once [close] is called, no other methods should be called.
+  void close() => _client.close();
```

Comment:
Q: is this a common convention in Dart?

follow up: and is the motivation to eagerly garbage collect things?

Reply:
If running on the comand-line VM we'll leave the VM running while there are open ports (short of an explicit call to system exit). This will close the http socket so the VM won't delay at shutdown; https://pub.dev/packages/http#using.

Reply:
Well, it's only concerning if you think about it :)

it is typical for Dart however, perhaps because we've always been single threaded, so do async work using futures. The bigger issue tends to be delaying VM exit rather than interrupting actual work.

@natebosch - I see that we didn't expose `close()` for `package:google_generative_ai` - thoughts on doing it for the generated clients here?